src/main_controller.ts

```diff
@@ -112,13 +112,14 @@ export class MainController implements vscode.Disposable {
 
         this.disposables.push(
             vscode.commands.registerCommand("_getNeovimClient", () => this.client),
-            vscode.commands.registerCommand("vscode-neovim.lua", async (lua) => {
-                if (!lua) {
+            vscode.commands.registerCommand("vscode-neovim.lua", async (...codes: (string | string[])[]) => {
+                const luaCode = codes.flat().join("\n");
```

Comment:
it's probably less surprising if each chunk is run separately. joining them could result in unintentionally shadowed variables, etc.

Reply:
Or use this one:
```typescript
            vscode.commands.registerCommand("vscode-neovim.lua", async (code: string | string[]) => {
                const luaCode = typeof code === "string" ? code : code.join("\n");
```

Reply:
The first one might be better because commands might have different parameter configurations in different setups. For example, with `arg = ['foo', 'bar']`, the command could be called as `cmd(arg)` or `cmd(...arg)`

Reply:
This is why variadic parameters (`...foo`) are often an anti-feature. They add useless ambiguity for very little gain. They should rarely be used, except when there is a clear purpose.

Reply:
I'm going to merge it. This command is for users to use in vscode. There may be some other extensions that need to call the command, such as multi-commands.

Reply:
> I'm going to merge it. This command is for users to use in vscode. There may be some other extensions that need to call the command, such as multi-commands.

why does that imply the need for a variadic parameter? how would the usage look like with and without a variadic parameter?

Reply:
Just make this command a little more flexible. We don't know how users will call it. Maybe changing it to `(...code: string[])` can be compatible with most use cases

Reply:
why is that worth the ambiguity that we identified above? 

> We don't know how users will call it.

we define how it should be called.